lib/MusicBrainz/Server/Data/URL.pm

```diff
@@ -309,21 +309,29 @@ sub _merge_impl
 
 sub find_by_url {
     my ($self, $url) = @_;
+
     my $normalized = URI->new($url)->canonical;
-    my $query = 'SELECT ' . $self->_columns . ' FROM ' . $self->_table .
-                ' WHERE url = ?';
-    $self->query_to_list($query, [$normalized]);
+    my $row = $self->c->prefer_ro_sql->select_single_row_hash(
+        <<~"SQL",
+        SELECT ${\($self->_columns)}
+          FROM ${\($self->_table)}
+         WHERE url = ?
+        SQL
+        $normalized,
+    );
+    if (defined $row) {
+        return $self->_new_from_row($row);
+    }
```

Comment:
I'm going to be annoying, but if `find_by_url` always returns one result, shouldn't it also be renamed to `get_by_url`? :stuck_out_tongue_closed_eyes: `find_by_` methods seem to mostly return lists.

Reply:
Fair point, I made the change. :)